controllers/targetnamespace_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// +build all
+
+package controllers
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/gobuffalo/envy"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	v1alpha1 "github.com/Azure/azure-service-operator/api/v1alpha1"
+	"github.com/Azure/azure-service-operator/pkg/helpers"
+)
+
+func TestTargetNamespaces(t *testing.T) {
+	t.Parallel()
+	defer PanicRecover(t)
+	ctx := context.Background()
+
+	// Check that resources in default and watched get reconciled
+	// successfully, but ones created in other ones don't.
+	rgName := tc.resourceGroupName
+	rgLocation := tc.resourceGroupLocation
+
+	newName := func() string {
+		return "storageacct" + helpers.RandomString(6)
+	}
+
+	createNamespaces(ctx, t, "watched", "unwatched")
+
+	configuredNamespaces := envy.Get("AZURE_TARGET_NAMESPACES", "")
+
+	instance := v1alpha1.StorageAccount{
```

Comment:
rather than `instance` and `instance2`, instead use more clear names like `instanceDefaultNamespace` and `instanceWatchedNamespace` ?